src/librustc_typeck/mem_categorization.rs

```diff
@@ -74,22 +74,24 @@ pub enum PlaceBase {
 }
 
 #[derive(Clone, Debug)]
-pub enum Projection<'tcx> {
+pub enum ProjectionKind<'tcx> {
     /// A dereference of a pointer, reference or `Box<T>` of the given type
     Deref(Ty<'tcx>),
     /// An index or a field
     Other,
 }
 
+#[derive(Clone, Debug)]
+pub struct Projection<'tcx> {
```

Comment:
I thought we also wanted to introduce `ty` in the Projection structure (https://github.com/rust-lang/project-rfc-2229/blob/master/hir-place-target.md). Is that not the case anymore or will this change be introduced in a later PR?

Reply:
That is part  https://github.com/rust-lang/project-rfc-2229/issues/5

Reply:
If we do that here, would've have to make sure at each step the type information gets read and put in place. Keeps this PR very focused and allows us to still do more work in parallel later on. 

I think after this gets merged we can parallely work on:
- https://github.com/rust-lang/project-rfc-2229/issues/5
- https://github.com/rust-lang/project-rfc-2229/issues/2
- https://github.com/rust-lang/project-rfc-2229/issues/7

Reply:
Sounds good, just wanted to confirm. Since I worked on this PR too, I'll let someone else approve it :)

Reply:
That's why I assigned this to @ChrisPardy and @null-sleep, since they attended the meeting about the refactor but @sexxi-goose/rustaceans feel free to approve it :)